include/libcyphal/application/registry/register.hpp

```diff
@@ -0,0 +1,188 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_APPLICATION_REGISTRY_REGISTER_HPP_INCLUDED
+#define LIBCYPHAL_APPLICATION_REGISTRY_REGISTER_HPP_INCLUDED
+
+#include "libcyphal/common/cavl/cavl.hpp"
+#include "libcyphal/common/crc.hpp"
+#include "registry_value.hpp"
+
+#include <cstdint>
+
+namespace libcyphal
+{
+namespace application
+{
+namespace registry
+{
+
+/// Defines possible errors that can occur when setting a register value.
+///
+enum class SetError : std::uint8_t
+{
+    Existence,   /// The register does not exist.
+    Mutability,  /// Register is immutable.
+    Coercion,    /// Value cannot be coerced to the register type.
+    Semantics,   /// Rejected by the register semantics (e.g., out of range, inappropriate value, bad state, etc.).
+};
+
+/// Defines interface for a register.
+///
+/// NOSONAR cpp:S4963 for below `class IRegister` - protected destructor is necessary.
+///
+class IRegister : public cavl::Node<IRegister>  // NOSONAR cpp:S4963
+{
+public:
+    /// Defines the type of the register name.
+    ///
+    using Name = const char*;
+
+    /// Defines behavior flags of the register value.
+    ///
+    struct Flags final
+    {
+        /// True if the register value can be changed.
+        bool mutable_{false};
+
+        /// True if the register value is retained across application restarts.
+        bool persistent_{false};
+    };
+
+    /// Defines a pair of the register value and its flags.
+    ///
+    struct ValueAndFlags final
+    {
+        Value value_;
+        Flags flags_;
```

Comment:
Public fields should not have the underscore suffix

Reply:
will fix in the next pr